.github/workflows/auto-supabase-deploy.yml

```diff
@@ -0,0 +1,26 @@
+name: Deploy Supabase Migrations on Main Merge
+
+on:
+  push:
```

Comment:
This need to be done in build workflow already existing not new one.
Just before `Install Supabase CLI` i think it's the best place